### PR TITLE
[ACTIONS & SOURCES] Spotify new actions

### PR DESCRIPTION
**Pull request by vellames-turing**
**Is there a specific app this action is for?**
Spotify

Please provide a link to the relevant API docs for the specific service / operation.
https://developer.spotify.com/documentation/web-api/reference/

**Actions**

- [x] Create Playlist | Create a new playlist
- [x] Save Track | Save a track to "Your Music" library
- [x] Add Track to Playlist | Add a track to one of your playlists
- [x] Get Audio Features for Track | Get the audio features (duration, tempo, etc.) for a track
- [x] Find Track | Look up a track by title/artist or by entering a track ID
- [x] Artist's Top Tracks | Display artist's top 10 tracks
- [x] Get List of Category's Playlists in Genres & Moods | Retrieve playlists in each specific category in Genres & Moods
- [x] Get Playlist's List of Tracks | Retrieve a list of all tracks in a specified playlist
- [x] Remove Track | Delete the track with the given ID
- [x] Remove Track from Playlist | Delete the track with the given ID from a playlist

**Addictional Improvements**

- [x] Change app from CommonJS to ES Module
- [x] Change New Playlist Source from CommonJS to ES Module
- [x] Change New Saved Track Source from CommonJS to ES Module
- [x] Change New Track In Playlist Source from CommonJS to ES Module
- [x] Fixed a bug on New Track In Playlist Source. When a user added the same music to a distinct playlists, just one event was triggered

### CODE REVIEW

**Comment by dylburger** [comment]
- [ ] [Create components to address specific use cases whenever possible](https://pipedream.com/docs/components/guidelines/#component-scope)
- [ ] [Component `key`s should follow the format `app_name_slug-slugified-component-name`](https://pipedream.com/docs/components/guidelines/#component-key-pattern)
- [ ] Components should follow the [standard directory structure](https://pipedream.com/docs/components/guidelines/#folder-structure)
- [ ] [Prefer Node.js client libraries to REST APIs](https://pipedream.com/docs/components/guidelines/#using-apis-vs-client-libraries)
- [ ] When making API requests, [handle pagination](https://pipedream.com/docs/components/guidelines/#pagination) to ensure all data / events are processed
- [ ] [Use `secret` props](https://pipedream.com/docs/components/guidelines/#capturing-sensitive-data) to capture sensitive data
- [ ] Props and methods should be defined in [app files](https://pipedream.com/docs/components/guidelines/#app-files) whenever possible
- [ ] Document methods with [JS Docs](https://pipedream.com/docs/components/guidelines/#methods)
- [ ] [Use `optional` props](https://pipedream.com/docs/components/guidelines/#optional-vs-required-props) whenever possible, and [set a `default` value](https://pipedream.com/docs/components/guidelines/#default-values) where you can
- [ ] Use [async options](https://pipedream.com/docs/components/guidelines/#async-options) to accept user input wherever possible
- [ ] [Create subscription webhooks sources (vs polling sources) whenever possible](https://pipedream.com/docs/components/guidelines/#webhook-vs-polling-sources)
- [ ] [Always emit a `summary` with events](https://pipedream.com/docs/components/guidelines/#emit-a-summary)
- [ ] [Use a built-in deduping strategy wherever possible](https://pipedream.com/docs/components/guidelines/#deduping)
- [ ] When building webhook sources, [follow these best practices](https://pipedream.com/docs/components/guidelines/#webhook-sources)
- [ ] When building polling soures, [follow these best practices](https://pipedream.com/docs/components/guidelines/#polling-sources)

File: components/spotify/actions/add-items-to-playlist/add-items-to-playlist.mjs [hunk 1]
```diff
@@ -0,0 +1,53 @@
+import lodash from "lodash";
+import spotify from "../../spotify.app.mjs";
+
+export default {
+  name: "Add items to a playlist",
```
Comment (dylburger) [inline components/spotify/actions/add-items-to-playlist/add-items-to-playlist.mjs:5]: I normally try to match the case of the operation in the API docs. [In Spotify's case](https://developer.spotify.com/documentation/web-api/reference/#category-playlists) this is: "Add Items to a Playlist"

**Comment by dannyroosevelt** [comment]
> @dannyroosevelt Yeah. Looks like a bug. I mentioned this before on Slack. If you click on save or deploy the changes are commited. This happens only if `useQuery` is true. Do you want me to open a new issue?

Actually I ran into an entirely separate bug that has since been fixed. I see the one you're referencing though, and Dylan has it on his radar.

**Comment by dannyroosevelt** [comment]
@vellames-turing everything is working for me -- nice job! I just pushed a commit with a bunch of 1-line tweaks to most of the components for a new $export we're adding to the component API (not yet documented...I'm working on that).

$summary will be used to display a human readable success message in the soon-to-be-released new version of our workflow builder. Can you please review my changes and let me know if you have comments / edits, etc?

**Pull request merged**